### ISSUE TITLE
English verbs wrongly recognized as nouns in particular context

### ISSUE THREAD

**Issue by evanmiltenburg** [explosion/spaCy#1403]
It seems that all verbs ending with *-ing* that aren't preceded by an auxiliary (*is X-ing*) are wrongly classified as nouns in the following examples:

* A boy in a black swimsuit playing near the water.
* One man kite surfing in the ocean.
* Men throwing tomatoes from a large trash bin.
* Child playing in a pool with clean water.
* A man in a uniform saluting.

Now these sentences do look a bit unusual, but embedding them doesn't help either:

* I saw a child playing in a pool with clean water.

Output: `[('I', 'PRON'), ('saw', 'VERB'), ('a', 'DET'), ('child', 'NOUN'), ('playing', 'NOUN'), ('in', 'ADP'), ('a', 'DET'), ('pool', 'NOUN'), ('with', 'ADP'), ('clean', 'ADJ'), ('water', 'NOUN'), ('.', 'PUNCT')]`

Here, *playing* should be a verb, but sentences like these are consistently mis-tagged.

```python
>>> import spacy
>>> nlp = spacy.load('en')
>>> doc = nlp('A boy in a black swimsuit playing near the water.')
>>> [(tok.orth_, tok.pos_) for tok in doc]

[('A', 'DET'), ('boy', 'NOUN'), ('in', 'ADP'), ('a', 'DET'), ('black', 'ADJ'), ('swimsuit', 'NOUN'), ('playing', 'NOUN'), ('near', 'ADP'), ('the', 'DET'), ('water', 'NOUN'), ('.', 'PUNCT')]
```
## Info about spaCy

* **spaCy version:** 1.9.0
* **Platform:** Darwin-16.7.0-x86_64-i386-64bit
* **Python version:** 3.6.1
* **Installed models:** __cache__, de, en, en-1.1.0

**Comment by honnibal** [comment]
The `en_core_web_sm` model in the 1.x family has been pruned with very aggressive L1 regularization. That's why the parser is 50mb, while the `en_core_web_md` model is more like 500mb. Given that the training data is pretty inconsistent about JJ vs VBG annotations, I guess it makes sense that the model "gives up" on difficult VBG cases in order to reduce the number of parameters.

If you're using the `en_core_web_md` model...Well, that's more surprising. You might try post-processing the POS tags by reference to the dependency parse, which might be more correct.

In any case, the v2 models should have notably higher tag accuracy. Have a look here: https://alpha.spacy.io/models/

**Comment by evanmiltenburg** [comment]
Thanks for the hints! I just installed `en_core_web_md`, and it still gets 4/5 wrong. I'll look at the v2 models.

**Comment by honnibal** [comment]
You might want to reference `en_core_web_sm` explicitly, to make sure the linking isn't playing funny tricks: `nlp = spacy.load('en_core_web_md')`

**Comment by evanmiltenburg** [comment]
Yes, I tried exactly that:

```python
>>> import spacy
>>> nlp = spacy.load('en_core_web_md')
>>> doc = nlp('A boy in a black swimsuit playing near the water.')
>>> [(tok.orth_, tok.pos_) for tok in doc]
[('A', 'DET'), ('boy', 'NOUN'), ('in', 'ADP'), ('a', 'DET'), ('black', 'ADJ'), ('swimsuit', 'NOUN'), ('playing', 'NOUN'), ('near', 'ADP'), ('the', 'DET'), ('water', 'NOUN'), ('.', 'PUNCT')]
```

**Comment by honnibal** [comment]
Results with 2:

```
>>> doc = nlp('A boy in a black swimsuit playing near the water.')
>>> [(w.text, w.pos_) for w in doc]
[('A', 'DET'), ('boy', 'NOUN'), ('in', 'ADP'), ('a', 'DET'), ('black', 'ADJ'), ('swimsuit', 'NOUN'), ('playing', 'VERB'), ('near', 'ADP'), ('the', 'DET'), ('water', 'NOUN'), ('.', 'PUNCT')]
```

Seems good.

**Comment by evanmiltenburg** [comment]
Cool, we found a *qualitative* improvement in the output! Was this with `en_core_web_sm` or `en_core_web_lg`?

**Comment by honnibal** [comment]
`en_core_web_sm`. I tested on v2.0.0a15 because that's what happened to be in my virtualenv, but the same should be true of v2.0.0a16 as well, in both sm and lg.

Are you doing the Visual Question Answering task?

**Comment by evanmiltenburg** [comment]
Close! I work on automatic image description. We're currently analyzing different verbs that are used to describe actions. But a large amount of descriptions in our corpus has the same structure of the examples I gave above. So it's essential that they are parsed correctly.

**Comment by honnibal** [comment]
Here's the tagger implementation in case it helps:

https://github.com/explosion/spaCy/blob/develop/spacy/pipeline.pyx#L328

Model:

https://github.com/explosion/spaCy/blob/develop/spacy/_ml.py#L312

It's a 4-layer CNN, where the input representation is constructed by concatenting several features for each word, which are then mixed with a 1-layer MLP. The features are lower-cased string, prefix, suffix and word shape in `en_core_web_sm`. The `en_core_web_lg` model makes use of pre-trained GloVe vectors as a feature as well. That's currently the only difference between them, although this could change.

If you're doing qualitative analysis and just want more accuracy, I suppose you could try training with `DEP` and `HEAD` features --- hopefully you see how to do that; simply add the attribute IDs to the `cols` array, and make extra embedding tables for them. `HEAD` provides a numeric offset value, so you can actually get sensible results from including it as a dimension explicitly. Alternatively clipping it at 4 and -4 and embedding it into a couple of dimensions is good.
